Add Compatibility for Lib's Disguises and CMI Vanish & Nameplates Hiding Feature

g2213swo opened the pull request:
### Description:

In this pull request, I've worked on ensuring compatibility between Lib's Disguises and CMI's vanish functionality. Initial testing indicates that the implementation is successful.

### Changes:

1. Introduced a boolean variable `hideNameplate` in `NamedEntityManager.java` along with its getter methods.
2. Implemented a feature to hide nameplates when a player is in vanish mode and invisibility potion effect

### Note:
I'm suggesting that the ability to toggle the nameplate hiding feature be given to server administrators. However, the toggle functionality has yet to be added.

Feedback and reviews would be greatly appreciated. Thank you!

Xiao-MoMi commented on src/main/java/net/momirealms/customnameplates/object/carrier/NamedEntityManager.java:
```diff
@@ -77,6 +78,7 @@ public boolean isNearby(Player viewer) {
     }
 
     public void spawn(Player viewer) {
+        if (hideNameplate) return;
```
会与正在隐身的玩家想要预览自己的铭牌冲突

Xiao-MoMi reviewed (changes requested):
resolve preview conflict

g2213swo commented:
> resolve preview conflict

I removed the original method and added a judgment in the method for sending data packets to other players. I'm not sure if this works...
